src/index.ts

```diff
@@ -1,3 +1,71 @@
 import './polyfills'
 import './components/shop'
 import './components/checkout-button'
+import {
+  ShopConfiguartion,
+  Product,
+  CheckoutOptions,
+  Order,
+} from './types/shop'
+import { Storage } from './storage'
+
+export class Shop {
+  private configuration: ShopConfiguartion
+  private hubApi: HubApi
+  private storage: Storage
+
+  constructor(configuration: ShopConfiguartion) {
+    this.configuration = configuration
+    this.hubApi = new HubApi(
+      configuration.hubUrl
+        ? configuration.hubUrl
+        : `https://hub.nimiq${configuration.live ? '' : '-testnet'}.com`,
+    )
+    this.storage = new Storage(
+      configuration.id,
+      configuration.live,
+      configuration.live,
+    )
+  }
+
+  // TODO(svub) Q: Meta data is to be freely defined by shop owner. String or Object? (will be run through JSON.stringify anyhow)
+  async checkout(products: Product[], meta: string): string {
+    const price =
+      products
+        .map(product => product.price)
+        .reduce((sum, price) => sum + price) * 1e5
+
+    const signedTx = await this.pay(price)
+    const orderId = await this.order(products, meta, signedTx)
+
+    return orderId
+  }
+
+  private async pay(price: number) {
+    const options: CheckoutOptions = {
+      appName: this.configuration.name,
+      recipient: this.configuration.address,
+      value: price,
+      fee: this.configuration.fee,
+      shopLogoUrl: this.configuration.logo,
+    }
+    // TODO(svub) what type is `signedTx`? (also change below)
+    const signedTx = await this.hubApi.checkout(options)
+    return signedTx
+  }
+
+  private async order(
+    products: Product[],
+    meta: string,
+    signedTx: any,
```

Comment:
Could we use another type than `any`?

Reply:
yes, I was wondering what type it is. For me, VS does not seem to load the Nimiq.d.ts, so I don't see any suggestions. :(

Reply:
See updated comment. I found the type. 
https://github.com/nimiq/hub/blob/574adcf5880c150b7b9d3cb016aeea78034c1316/src/lib/PublicRequestTypes.ts#L44

But how to load/import it?

Reply:
found solution and pushed changes. I think it's not a really clean solution, let me know if there's a better one.